### PR TITLE
Vcpkg sample

### PR DESCRIPTION
**Pull request by vhvb1989**
Adding a sample for how to make an application using vcpkg to get the Azure sdk for C++ libs

fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/1388

### CODE REVIEW

**Review by RickWinter — COMMENTED** [review]
The file changes here looks unrelated to the issue/title of the PR.

File: samples/integration/vcpkg/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# base image installs keyvault beta1 from vcpkg only
+FROM vhvb1989/keyvault:beta1
+
+RUN cd vcpkg \
+    && sudo ./vcpkg install azure-identity-cpp
```
Comment (ahsonkhan) [inline samples/integration/vcpkg/.devcontainer/Dockerfile:5]: Just this? I think we should use storage as an example instead.

File: samples/integration/vcpkg/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# SPDX-License-Identifier: MIT
+
+# VCPKG integration.
+if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+    CACHE STRING "")
+endif()
+if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
+  set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}" CACHE STRING "")
+endif()
```
Comment (ahsonkhan) [inline samples/integration/vcpkg/CMakeLists.txt:11]: This file is meant for users to copy/paste and use in their applications. We should try to simplify it as much as possible. Is this actually needed?

File: sdk/core/azure-core/test/ut/md5.cpp [hunk 3]
```diff
@@ -118,3 +118,33 @@ TEST(Md5Hash, CtorDtor)
     Md5Hash instance;
   }
 }
+
+#include <thread>
+TEST(Md5Hash, multiThread)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/test/ut/md5.cpp:123]: Revert the change to this file.

File: samples/integration/vcpkg/src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+#include <azure/core.hpp>
+#include <azure/identity.hpp>
+#include <azure/keyvault/key_vault.hpp>
+
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <thread>
+
+using namespace Azure::Security::KeyVault::Keys;
```
Comment (ahsonkhan) [inline samples/integration/vcpkg/src/main.cpp:24]: Since KeyVault isn't GA'ing yet, let's use a storage upload/download blob sample instead.

Comment (ahsonkhan) [reply samples/integration/vcpkg/src/main.cpp:24]: Also, the sample seems quite busy, can we simplify it a bit? We want this as a quick starting point for customers to copy/paste and get started. We don't need it to do too much.

File: samples/integration/vcpkg/.devcontainer/Dockerfile (hunk 1 above)
Comment (vhvb1989) [reply samples/integration/vcpkg/.devcontainer/Dockerfile:5]: I can add another sample for storage.  why not.

File: samples/integration/vcpkg/CMakeLists.txt (hunk 2 above)
Comment (vhvb1989) [reply samples/integration/vcpkg/CMakeLists.txt:11]: The expected flow to test/play this samples is:

- Clone the c++ repo.
- Open a sample folder
- build/run

And that's it.  Without setting vcpkg path, toolchain or anything. That's as simplest as I am trying to make this.

Then, once it's building/running, people can start looking at the code for things to copy/paste.

File: samples/integration/vcpkg/.devcontainer/Dockerfile (hunk 1 above)
Comment (vhvb1989) [reply samples/integration/vcpkg/.devcontainer/Dockerfile]: Added storage sample as well

File: samples/integration/vcpkg-keyvault/.devcontainer/devcontainer.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+		"streetsidesoftware.code-spell-checker",
+		"ms-vscode.cpptools",
+		"xaver.clang-format",
+		"twxs.cmake",
+		"ms-vscode.cmake-tools",
+		"eamodio.gitlens",
+		"davidschuldenfrei.gtest-adapter"
+	],
+    // Do not run as root. See https://aka.ms/vscode-remote/containers/non-root.
+    "remoteUser": "azure-sdk-for-cpp"
```
Comment (RickWinter) [inline samples/integration/vcpkg-keyvault/.devcontainer/devcontainer.json:19]: Simplier and easier name to remember
```suggestion
    "remoteUser": "azure-user"
```

Comment (vhvb1989) [reply samples/integration/vcpkg-keyvault/.devcontainer/devcontainer.json:19]: There is  relation between the image that is pulled from docker hub (vhvb1989/keyvault:beta1) and this value.

The image has created a linux user with this name already.  Changing the name would break vscode workspace to work, until the image from the docker hub is updated.   

And the user name is not really such relevant!  What we want is a ready-to-use dev env where to run the samples
